source/views/directory/types.ts

```diff
@@ -0,0 +1,70 @@
+export type CampusLocation = {
+	display: string
+	buildingabbr: string
+	building: string
+	phone: string
+	room: number
+	// added from ccc-server
+	shortLocation: string
+}
+
+export type Department = {
+	href: string
+	name: string
+}
+
+type Address = {
+	zip: string
+	city: string
+	country: string
+	state: string
+	street: Array<string>
+}
+
+export type OfficeHours = {
+	display: string
+	prefix: string
+	hrefLabel: string | null
+	href: string | null
+	content: string
+	// added from ccc-server
+	description: string
+	title: string
+}
+
+type OnLeave = {
+	start: string
+	end: string
+	type: string
+}
+
+export type DirectoryItem = {
+	campusLocations: Array<CampusLocation>
+	classYear: string | null
+	departments: Array<Department>
+	displayName: string
+	displayTitle: string | null
+	email: string | null
+	firstName: string
+	homeAddress: Address
+	homePhone: string | null
+	lastName: string
+	officeHours: OfficeHours | null
+	onLeave: OnLeave | null
+	photo: string
+	profileUrl: string | null
+	suffixName: string | null
+	thumbnail: string
+	title: string | null
+	username: string | null
+	// added from ccc-server
```

Comment:
We're not proxying through ccc, are we?

Reply:
I had proposed a changeset to ccc to proxy before we realized we'd end up losing results that way. The proxied endpoint never was merged, but I can take a look to see if I forgot something on the client side. Good catch.

Reply:
We're good, looks like the comment remained but the logic is all on the client.

```tsx
#  https://github.com/StoDevX/AAO-React-Native/blob/directory/source/views/directory/helpers.ts#L80-L83

const shortRoomDetail = (loc: CampusLocation): string => {
	const shortRoom = `${loc.buildingabbr} ${loc.room}`.trim()
	return `${shortRoom ? `${shortRoom} • ` : ''}${loc.phone}`
}

# https://github.com/StoDevX/AAO-React-Native/blob/directory/source/views/directory/helpers.ts#L107-L110

const locations = campusLocations.map((loc: CampusLocation) => ({
	...loc,
	shortLocation: shortRoomDetail(loc),
}))
```